react/components/ProductSummary.js

```diff
@@ -133,19 +152,17 @@ const ProductSummaryCustom = ({ product, actionOnClick, children, href }) => {
 
   return (
     <ProductSummaryContext.Provider value={oldContextProps}>
-      <ProductContextProvider product={product} query={{ skuId }}>
-        <section
-          className={containerClasses}
-          onMouseEnter={handleMouseEnter}
-          onMouseLeave={handleMouseLeave}
-          style={{ maxWidth: PRODUCT_SUMMARY_MAX_WIDTH }}
-          ref={inViewRef}
-        >
-          <Link className={linkClasses} {...linkProps} onClick={actionOnClick}>
-            <article className={summaryClasses}>{children}</article>
-          </Link>
-        </section>
-      </ProductContextProvider>
+      <section
```

Comment:
Why there's no ProductContextProvider here? This was moved to the `ProductSummaryWrapper`, is that it?

Reply:
Yes, I moved the `ProductContextProvider ` to the  `ProductSummaryWrapper `. I did this because I needed to use a hook from the ProductContextProvider.

But after reading your comments, I don't think it is the best approach. I refactored the code. Now there is a new component called `ProductPriceSimulation`. I think it's a way better now. What do you think?